streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -337,13 +347,24 @@ public void close(final Map<TopicPartition, Long> offsets) throws IOException {
     @Override
     public void checkpoint(final Map<TopicPartition, Long> offsets) {
         checkpointableOffsets.putAll(offsets);
-        if (!checkpointableOffsets.isEmpty()) {
-            try {
-                checkpoint.write(checkpointableOffsets);
-            } catch (final IOException e) {
-                log.warn("Failed to write offset checkpoint file to {} for global stores: {}", checkpoint, e);
+
+        final Map<TopicPartition, Long> filteredOffsets = new HashMap<>();
+
+        // Skip non persistent store
+        for (final Map.Entry<TopicPartition, Long> topicPartitionOffset : checkpointableOffsets.entrySet()) {
+            final String topic = topicPartitionOffset.getKey().topic();
+            if (globalNonPersistentStoresTopics.contains(topic)) {
+                filteredOffsets.remove(topicPartitionOffset.getKey());
```

Comment:
Do we need to call remove ever? Since `filteredOffsets` is constructed empty can we just do the following:

```
if (!globalNonPersistentStoresTopics.contains(topic)) {
    filteredOffsets.put(topicPartitionOffset.getKey(), topicPartitionOffset.getValue());
}
```

Reply:
Makes sense.